### DISCUSSION TITLE
A2 hosting

### DISCUSSION THREAD

**Discussion by rossi-jeff** [sveltejs/kit#9282] [General]
I have my domain and several node apps svelte, react, vue, angular, and nestjs at A2. I have had no success with latest svelte kit with multiple +server.ts files using node adapter. Does anyone have any pointers or is deploying svelte kit on cpanel hosting a lost cause?

**Comment by dev-itzarun** [comment]
@rossi-jeff have you succeed in deploying ?

**Reply by amirbenasr** [reply]
@dev-itzarun have you yet ?

**Reply by dev-itzarun** [reply]
@amirbenasr No , using aws and vercel now.

If you have a vps in a2hosting you can deploy directly as node app , it will works fine I think.

The actual issue is with shared hosting right ?

If you have a shared or cloud , i think we have to set up a reverse proxy in #a2hosting to make this work.

I have **wasted** 2 weeks trying to make this work in a shared hosting (that support node).

If you are building a CSR project use static adapter..

Static adapter works fine for me all hosting services.

**Reply by amir-dragon** [reply]
I succeded on deploying it as a node server and it works just fine, i tried the static adapter, but it won't allow me since i have api/endpoints routes 😅

**Reply by dev-itzarun** [reply]
@amirbenasr, that whats iam also trying to say. When you have only CSR, you can use static adapter, and It will work , but if you have an SSR function, you have to use the adapter-node, and it's not working.

But usually shared servers also allow us to host SSR apps with node, right ? But it's not working with Svelte Kit.

The 'setup node app' option in Cpanel allows us to set the root path , env , etc...

But we can't even run the build.

I currently do not have any shared servers.

So I can't share any screenshots or logs now...

**Reply by amirbenasr** [reply]
@dev-itzarun Let me tell you, that I was able to run Sveltekit app on shared hosting with cpanel CI/CD :) 
It was a little bit tricky but here are the steps I've made to help you if you wanted to : 
-clone the repository that you're working on from the git version control tab.
-if the run build is not working  ( make sure your NODE_ENV is not set to production )
- if the sveltekit app is using node-adapter run these comands:

1. npm install --save-dev ( this will force the npm to install the devDependencies which will allow you to run : npm run build )
2. npm run build
3. install pm2 and run the node at build/index.js ( if you want to specify a port add PORT=XXXX before the following command)
-create a .htaccess file inside the domain | subdomain, and configure it to run on specific ports ( look for the .htaccess file for the nodejs app )
-And voila you get it :) Hope this helps you

**Comment by thomascayne** [comment]
Hey @amirbenasr like @dev-itzarun I have been struggled for longer than 2 weeks to get sveltekit app to run on a2hosting. Support is not helping and so difficult to find solution. Also, the ssh keys doesn't work in github actions. When I use Secrets the ping from the terminal shows permission denied but when i put the same ssh private key in Deploy Keys permission granted but Deploy Keys seems inaccessible in Github Actions.

Nevertheless, I tried following your steps to manually set up the app but cannot get it to work. Even the npm install from cPanel gave error. It's shared hosting.

Can you outline a more detailed step? I am sure since Sveltekit is hot right now others, including myself, will benefit from this.

Cheers!

**Reply by dev-itzarun** [reply]
@thomascayne the only solution is to use vps plans use adapter-node 

Or consider using vercel / railway.

**Reply by amirbenasr** [reply]
@thomascayne The issue with shared hosting and cpanels IIRC is the amount of garbage you need to deal with just to make the app work s in a static level, for Sveltekit the only solution I had is by using the static adapter but for SSR and spa I won't recommend you using shared hosting because it's just a pain in the ASS

**Reply by amirbenasr** [reply]
I recommend you using AWS for hosting or Vercel ( they just deal with that garbage for you )

**Comment by rossi-jeff** [comment]
I used static builds for most of the node applications I have hosted. Fo
backend nestjs seems to work ok.

On Wed, Dec 6, 2023 at 8:44 AM Arun Kumar ***@***.***> wrote:

> @thomascayne <https://github.com/thomascayne> the only solution is to use
> vps plans.
>
> Or consider using vercel / railway.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sveltejs/kit/discussions/9282#discussioncomment-7776472>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFD2WALSDCIBG4PEGGKDWLYIBZFFAVCNFSM6AAAAAAVNM77NKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONZWGQ3TE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Regards

Jeff Rossi

1506 Tuscaloosa Ave
Holly Hill, FL 32117
Home: (386) 226-8913
Cell: (386) 316-8485
